### PR TITLE
[TASK] Add SSL cert to rippled to enable secure websockets (DEC-22)

### PR DESCRIPTION
**Pull request by wilsonianb**

### CODE REVIEW

**Comment by wilsonianb** [comment]
+r

File: rippled.cfg [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
-ip = 0.0.0.0
+[port_wss_admin]
+port = 51237
+ip = 127.0.0.1
 admin = allow
-protocol = http
+protocol = wss
 
-[port_ws]
+[port_ws_public]
```
Comment (stevenzeiler) [inline rippled.cfg]: are we going to continue to support `ws` or require `wss`?

**Comment by tdfischer** [comment]
LGTM :+1:

File: rippled.cfg [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
 port = 51233
 ip = 0.0.0.0
-admin = allow
-protocol = ws
+protocol = wss, ws
 
 [websocket_secure]
 0
```
Comment (stevenzeiler) [inline rippled.cfg]: Should `websocket_secure` be set to 1?

File: rippled.cfg [hunk 3]
```diff
@@ -1,26 +1,34 @@
...
 
 [port_peer]
 port = 51235
 ip = 0.0.0.0
 protocol = peer
 
-[port_rpc]
-port = 51234
-ip = 0.0.0.0
+[port_wss_admin]
```
Comment (stevenzeiler) [inline rippled.cfg]: It looks like SSL is only available to admins rather than all clients.

File: rippled.cfg (hunk 1 above)
Comment (wilsonianb) [inline rippled.cfg]: Our mainnet client handlers support `ws`, but only allowing `wss` sounds sound.

**Comment by codius-montecarlo** [comment]
The build is good and I'm seeing 1 +1s from the following folks:

* tdfischer

As such, I'll attempt to merge and close this pull request.

**Comment by stevenzeiler** [comment]
Okay the merge is fine, but we have to add more to actually enable secure websockets for normal users in addition to admins.

File: rippled.cfg (hunk 2 above)
Comment (wilsonianb) [inline rippled.cfg]: I think that field is deprecated. I can remove it.

Comment (stevenzeiler) [inline rippled.cfg]: :+1:

File: rippled.cfg (hunk 3 above)
Comment (wilsonianb) [inline rippled.cfg]: `port_wss_admin` has `wss`
`port_ws_public` has `ws` and `wss`

Comment (stevenzeiler) [inline rippled.cfg]: Oh I see, misunderstood the cfg file

**Pull request merged**